src/actions/ast.js

```diff
@@ -61,15 +56,17 @@ export function setSymbols(sourceId: SourceId) {
   return async ({ dispatch, getState, sourceMaps }: ThunkArgs) => {
     const source = getSourceFromId(getState(), sourceId);
 
-    if (source.isWasm || hasSymbols(getState(), source) || !isLoaded(source)) {
+    if (source.isWasm && !isLoaded(source)) {
       return;
     }
 
-    await dispatch({
-      type: "SET_SYMBOLS",
-      sourceId,
-      [PROMISE]: getSymbols(sourceId)
-    });
+    if (!getSymbols(getState(), source)) {
+      await dispatch({
+        type: "SET_SYMBOLS",
+        sourceId,
+        [PROMISE]: parser.getSymbols(sourceId)
+      });
+    }
```

Comment:
This fixes the  failing tests, since we still want o fetch the extra content even if we don't reset the symbols.